x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/watcher/execution/WatchExecutionContext.java

```diff
@@ -85,6 +88,14 @@ public Watch watch() {
     public void ensureWatchExists(CheckedSupplier<Watch, Exception> supplier) throws Exception {
         if (watch == null) {
             watch = supplier.get();
+            // now that the watch exists, extract out the authentication
```

Comment:
this can be refactored into a small static method, so you could write also some tests (so many ifs/null checks to check). This could also be done lazily in the getter (not sure yet if that is a good idea though).

Reply:
This method could also be overwritten by sub classes (both happen to call `super.ensureExists()` though) - might be a good reason to load it lazily

Reply:
There looks to be no need to override this method, so ill make it final (and still move that bit to a static helper for testing)

Reply:
after some snooping I noticed we can and should do this #31926